app/src/org/commcare/tasks/DataPullTask.java

```diff
@@ -217,7 +219,7 @@ public void reportProgress(int progress) {
                         CommCareApplication._().releaseUserResourcesAndServices();
                     }
                     Logger.log(AndroidLogger.TYPE_USER, "Bad Auth Request for user!|" + username);
-                    return PullTaskResult.AUTH_FAILED;
+                    return new Pair<>(PullTaskResult.AUTH_FAILED, "");

```

Comment:
Good call! How about I do this in new PR and also update `SaveToDiskTask` to use the same object ([here](https://github.com/dimagi/commcare-odk/blob/master/app/src/org/commcare/tasks/SaveToDiskTask.java#L93-L95))